python/beeai_framework/retrieval/vector_stores/in_memory_vector_store.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from typing import Any
+
+try:
+    from langchain_core.documents import Document as LCDocument
+    from langchain_core.vectorstores import InMemoryVectorStore as LCInMemoryVectorStore
+
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
+        "Optional module [langchain] not found.\nRun 'pip install \"beeai-framework[langchain]\"' to install."
+    ) from e
+
+from beeai_framework.adapters.langchain.mappers.documents import lc_document_to_document
+from beeai_framework.adapters.langchain.mappers.lc_embedding import LangChainBeeAIEmbeddingModel
+from beeai_framework.backend.embedding import EmbeddingModel
+from beeai_framework.backend.types import DocumentWithScore
+
+
+class InMemoryVectorStore:
```

Comment:
I think you missed the inheritance part.

```
class InMemoryVectoreStore(VectoreStore):
```

Reply:
InMemoryVectoreStore is removed and the example is adapted to use dynamic loading